### ISSUE TITLE
Segfault on PurgeArchive

### ISSUE THREAD

**Issue by dennisjbell** [starkandwayne/shield#461]
`shieldd` was falling over, which was causing the vault to lock.  `/var/vcap/sys/log/shield/shieldd.log` contains the following stack trace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x5f93ff]

goroutine 34 [running]:
github.com/starkandwayne/shield/db.(*DB).PurgeArchive(0xc420054800, 0xc420156420, 0x10, 0x10, 0x2, 0xc420162f00)
        /tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/db/archives.go:286 +0x7f
github.com/starkandwayne/shield/core.(*Core).worker(0xc420064ea0, 0x4)
        /tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/core/core.go:714 +0x167a
created by github.com/starkandwayne/shield/core.(*Core).runWorkers
        /tmp/build/01f561ac/gopath/src/github.com/starkandwayne/shield/core/core.go:340 +0x151
```

**Comment by jhunt** [comment]
Which version of SHIELD is this?

**Comment by dennisjbell** [comment]
8.0.15

**Comment by jhunt** [comment]
This should be fixed in the 8.1.0 release, which should be out the door by next week (just in time for CF Summit Philly!)

**Comment by jhunt** [comment]
Fixed in v8.1.0 (https://github.com/starkandwayne/shield/releases/tag/v8.1.0)
